lib/appsignal/hooks/puma.rb

```diff
@@ -33,5 +37,59 @@ def stop_workers
         end
       end
     end
+
+    class PumaProbe
+      def initialize
+        @hostname = Appsignal.config[:hostname] || Socket.gethostname
+      end
+
+      def call
+        return unless ::Puma.stats
```

Comment:
if there a scenario where it returns `nil` as a value? other than in our tests

Reply:
Yes, right after boot the value is nill, it won't be populated until Puma fills the data:

* https://github.com/puma/puma/blob/master/lib/puma.rb#L16
* https://github.com/puma/puma/blob/ca03c520757e4ed1caab28bf4f60ec2ffe5b2585/lib/puma/launcher.rb#L91